File: Examples/test-suite/ruby/smart_pointer_overload_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# Author::    gga
# Copyright:: 2007
# License::   SWIG
#

require 'swig_assert'

require 'smart_pointer_overload'

include Smart_pointer_overload

f = Foo.new
b = Bar.new(f)


raise RuntimeError if f.test(3) != 1
raise RuntimeError if f.test(3.5) != 2
raise RuntimeError if f.test("hello") != 3

raise RuntimeError if b.test(3) != 1
raise RuntimeError if b.test(3.5) != 2
raise RuntimeError if b.test("hello") != 3

File: Examples/test-suite/lua/smart_pointer_overload_runme.lua
require("import")
import_to_globs("smart_pointer_overload")

f = Foo()
b = Bar(f)

assert(f:test(3) == 1)
--assert(f:test(3.5) == 2)	-- won't work due to being unable to overloads
assert(f:test("hello") == 3)

assert(b:test(3) == 1)
--assert(b:test(3.5) == 2)	-- won't work due to being unable to overloads
assert(b:test("hello") == 3)
